test/test-helpers.coffee

```diff
@@ -143,6 +143,27 @@ beforeEach ->
       @message = -> "Expected to emit #{jasmine.pp(expectedLog)}, actually emitted #{jasmine.pp(log)}"
       @env.equals_(expectedLog, log)
 
+    # Test with setTimeout not guaranteeing order. See
+    # https://github.com/rpominov/kefir/issues/134
+    toEmitOverShakyTime: (expectedLog, cb, timeLimit = 1000) ->
+      log = null
+      exports.withFakeTime (tick) =>
+        originalST = global.setTimeout
+        global.setTimeout = ->
+          if Math.random() > 0.5
```

Comment:
Not sure if randomness in tests is a good idea. What if we pass an option to `toEmitOverShakyTime` to tell what path to take?

```coffeescript
for reverse in [true, false]
  a = stream()
  expect(a.delay(0)).toEmitOverShakyTime reverse, [i, '<end>'], (tick) ->
    send(a, [i])
    send(a, ['<end>'])
```

```diff
- if Math.random() > 0.5
+ if reverse
```